### ISSUE TITLE
OAuth token error using Outlook REST API v2 on Mac with Gmail account type

### ISSUE THREAD

**Issue by sznorbert07** [OfficeDev/office-js#4117]
## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Mac
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Outlook, 16.81.2 (24012814), Mac
* Operating System: macOS Sonoma 14.2.1
* Email account with Gmail
 
## Current behavior
Calling https://outlook.office.com/api/v2.0/me/messages/ endpoint by providing the token which we get back from the office-js library through the Office.context.mailbox.getCallbackTokenAsync() method call causes a 401 response with the following error message:

```
{
    "error": {
        "code": "UnableToReadToken",
        "message": "OAuth token submitted with the request can not be parsed.",
        "innerError": {
            "oAuthEventOperationId": "98b76916-c783-4557-9290-e89784f83e74",
            "oAuthEventcV": "L99lnc8TOL0tgkYgUUQOZw.1",
            "errorUrl": "https://aka.ms/autherrors#error-InvalidToken",
            "requestId": "9d65df2f-13cf-bd38-2d82-462051440e67",
            "date": "2024-02-09T07:33:05"
        }
    }
}
```

This issue only happens while using Gmail account type from Outlook on Mac.
Our customers using the add-in on Mac with Gmail account started to get this error message from today morning (09/02/2024).

We know that Outlook REST v2 API is decommissioned and we must use Graph API from march 2024.
However for Gmail account we cannot use Graph API.
What is the recommended solution here to continue supporting add-in for our customers with Gmail account?

**Comment by sznorbert07** [comment]
I also have a question related to this topic:
Is it possible somehow to disable add-ins for Gmail accounts in Outlook on Mac via XML manifest?

**Comment by vtiti** [comment]
I have a related question: according to the https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/add-in-requirements#mail-server-requirements page the Outlook for Mac using with GMail accounts is not even supported anymore. Can I ask when it changed, or what is the official standing with these scenario? 
If it is not supported anymore why our addin is not disabled when our users selecting a GMail account?

**Comment by ajays-msft** [comment]
@sznorbert07 - Can you please help understand the impact, I mean how many users of your addin are effected?

**Comment by sznorbert07** [comment]
@ajays-msft - I do not know exactly how many people are affected by this issue, but it is a problem for anyone using Gmail on the Mac since Friday morning.

On the other hand, we have not been notified if there will be a breaking change for this scenario, nor have we been able to find any information about it, and we have already received several reports that our add-in does not work.
We have to deal with it somehow and come up with a clean solution as soon as possible to satisfy our customers' needs. If we cannot continue to support such a scenario by default (as it has worked in the past), we need to convince them why.

**Comment by derpadoo** [comment]
@sznorbert07 Might not be relevant to your use case, but the JSON Web Token payload returned from `Office.context.mailbox.getCallbackTokenAsync()` started changing January 1, 2024 (https://devblogs.microsoft.com/microsoft365dev/important-announcement-update-to-exchange-callback-tokens-for-outlook-add-ins/).

> If your add-in parses the token, or relies on anything inside the token string, it might be impacted.

**Comment by sznorbert07** [comment]
@derpadoo Yeah, it is not relevant for our use case. We do not rely on the JWT format.

**Comment by sznorbert07** [comment]
Meanwhile, I investigated whether we could use the Gmail API or not, but it is not possible, since the Mail ID we receive from office-js is in a different format than what Gmail uses.

I also checked whether I could use the Internet Message ID. For some of the emails I tested, I was able to access the email with the Gmail API. However, this raises the question of whether we can rely on it or not.

**Comment by neprasad-microsoft** [comment]
Hi @sznorbert07 
Can you verify if you get a token from Office.context.mailbox.getCallbackTokenAsync(), or is it undefined?

**Comment by sznorbert07** [comment]
> Office.context.mailbox.getCallbackTokenAsync()

Yes, I check it

**Comment by sznorbert07** [comment]
@neprasad-microsoft I have checked what you have asked.

Since getCallbackTokenAsync() has multiple overloads, I have tried the following calls:

1.  `Office.context.mailbox.getCallbackTokenAsync(result => console.log(result))`
2.  `Office.context.mailbox.getCallbackTokenAsync({ isRest: false}, result => console.log(result))`
3.  `Office.context.mailbox.getCallbackTokenAsync({ isRest: true}, result => console.log(result))`

The 1. and 2. calls are working and I get a token, but 3. doesn't.

Strange thing is that the token returned from the 1. or 2. call works perfectly with the REST API, however as per the documentation it clearly should be a token for the EWS.
Doc: https://learn.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-preview#outlook-office-mailbox-getcallbacktokenasync-member(1)

(Also, the documentation does not mention how it works on Outlook on Mac with Gmail account)

The 3. call however returns the following error JSON:
```
{"value":undefined,"status":"failed","error":{"name":"AccessRestricted","message":"Internal protocol error: '-2147467259'.","stack":undefined},"diagnostics":{"InvokeCodeResult":-2147467259}}
```
